spyder/config/tests/test_base.py

```diff
@@ -0,0 +1,51 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# Copyright (c) 2009- Spyder Project Contributors
+#
+# Distributed under the terms of the MIT License
+# (see spyder/__init__.py for details)
+# -----------------------------------------------------------------------------
+
+"""
+Tests for the spyder.config.base module.
+"""
+
+# Standard library imports
+import importlib
+import os.path as osp
+
+# Third party imports
+import pytest
+
+# Local imports
+import spyder.config.base
+
+
+# ============================================================================
+# ---- Tests
+# ============================================================================
+@pytest.mark.parametrize('version_input, expected_result', [
+    ('3.3.0', True), ('2', True), (('0', '5'), True), ('4.0.0b1', False),
+    ('3.3.2.dev0', False), ('beta', False), (('2', '0', 'alpha'), False)])
```

Comment:
This is quite confusing. Please don't abuse of `paremetrize` and simply test `is_stable_version` as it was tested before.

Reply:
I'm not sure how this can be fairly characterized as "abuse" considering it follows an essentially *identical* pattern to the canonical example of parameterization in the [Pytest docs](https://docs.pytest.org/en/latest/parametrize.html), and I find it clearer and much less code smelly than the pathological antipattern that is a long list of identical aside from input/expected assert statements. However, it is perhaps slightly less performant and adds extra lines to the logs for a relatively minor test, so I de-parameterized it as requested.

Reply:
> I'm not sure how this can be fairly characterized as "abuse"

Because there are lot of test cases packed up in three lines and it's hard to understand what's going there.

Reply:
"Abuse" is a pretty serious accusation, though though I know you didn't mean anything bad by it. Its perfectly valid for someone to find a certain pattern confusing and ask for it to be made clearer, which I hopefully did. However, I don't see how calling it "abuse" of a certain feature is really a fair characterization when its exactly how said feature is designed to be used (it was virtually *identical* to the very first example given in the docs, if you replace the input/expected values with these ones and ``eval()`` with ``is_stable_version``).